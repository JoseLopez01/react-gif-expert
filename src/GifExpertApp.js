import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Puch']);

  return (
    <>
      <h2>Gift Expert App</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories
          .map((category, i) => 
            (<GifGrid key={category} category={category} />)
          )
        }
      </ol>
    </>
  );
};
