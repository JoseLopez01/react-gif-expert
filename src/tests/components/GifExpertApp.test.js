import { shallow } from 'enzyme';
import { GifExpertApp } from '../../GifExpertApp';

describe('Probando GifExpertApp', () => {
  
  test('Debe crear bien el componente', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar una lista de categorias', () => {
    const categories = ['One Punch', 'Dragon Ball'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid')).toHaveLength(categories.length);
  })
});