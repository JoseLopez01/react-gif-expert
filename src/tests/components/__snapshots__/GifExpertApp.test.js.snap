// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando GifExpertApp Debe crear bien el componente 1`] = `
<Fragment>
  <h2>
    Gift Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Probando GifExpertApp Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gift Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
